### DISCUSSION TITLE
FIDI+Nordigen cannot handle transactions with amount of zero?

### DISCUSSION THREAD

**Discussion by welles** [firefly-iii/firefly-iii#6493] [Firefly III] [answered]
I have an account with a bank that waives your monthly fee if you meet certain conditions.
If the fee is waived, they still create a transaction to show you that nothing has been charged, which has an amount of "0.00".
Unfortunately FIDI seems unable to handle this and shows several error messages in each run.
This is the data that Nordigen shows for this transaction:
```
{
    "entryReference": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "bookingDate": "2022-10-04",
    "valueDate": "2022-10-01",
    "transactionAmount": {
        "amount": "0.00",
        "currency": "EUR"
    },
    "creditorAccount": {
        "iban": "0000000000" // Actual zeros
    },
    "debtorAccount": {
        "iban": "XXXXXXXXXX" // My account no
    },
    "remittanceInformationUnstructured": "Abrechnung 30.09.2022"
}
```

As you can see it has two problems: 1. The creditor account is all zeros, which seems to mean "it's us, your bank". 2. The amount is zero.

This results in the following errors from FIDI:
```
    Line #1: transactions.0.amount: Buchungsbetrag muss größer als 0 sein. (original value: "0.00")
    Line #1: transactions.0.source_id: Um fortzufahren, benötigen Sie eine gültige Quellkontenkennung und/oder einen gültigen Quellkontonamen. (original value: "(not found)")
    Line #1: transactions.0.source_name: Um fortzufahren, benötigen Sie eine gültige Quellkontenkennung und/oder einen gültigen Quellkontonamen. (original value: "(not found)")
```
I didn't want to create an issue because I figured I must not be the first person to run into this behavior. Is there a way to work around this?

**Comment by JC5** [comment, marked as answer]
Unfortunately, Firefly III will not accept transactions that are zero.

Would you like the data importer to filter them out beforehand?

**Reply by welles** [reply]
It probably shouldn't be default behavior, but maybe a `IGNORE_EMPTY_TRANSACTIONS` environment variable or something like it would be useful?

**Reply by JC5** [reply]
I think I'll make it default behavior anyway: these empty transactions would fail anyway when submitted to Firefly III. I'll make them warnings, that feels more user friendly.

**Reply by welles** [reply]
Sounds good, thank you!
